File: generated-components/reactNative/Card.js
import * as React from "react";
import { View, StyleSheet, Image, Text, Pressable } from "react-native";
import { useState } from "react";

function Card(props: any) {
  const [isHovered, setIsHovered] = useState(() => false);

  function handleMouseEnter() {
    if (props.hoverable) {
      setIsHovered(true);
    }
  }

  function handleMouseLeave() {
    if (props.hoverable) {
      setIsHovered(false);
    }
  }

  return (
    <Pressable
      style={cardStyle}
      onMouseEnter={(event) => handleMouseEnter()}
      onMouseLeave={(event) => handleMouseLeave()}
      onPress={(event) => props.onClick()}
    >
      {props.image ? (
        <Image
          source={{ uri: props.image }}
          alt={props.title || "Card image"}
          style={imageStyle}
        />
      ) : null}
      {props.title ? (
        <View style={titleStyle}>
          <Text>{props.title}</Text>
        </View>
      ) : null}
      {props.subtitle ? (
        <View style={subtitleStyle}>
          <Text>{props.subtitle}</Text>
        </View>
      ) : null}
      {props.children ? <View>{props.children}</View> : null}
    </Pressable>
  );
}

export default Card;

File: generated-components/swift/Card.swift
import SwiftUI

struct Card: View {
  let hoverable: Any
  let onClick: Any
  let className: Any
  let image: Any
  let title: Any
  let subtitle: Any
  let children: Any

  @State private var isHovered: Bool = false

  // Function state variables
  var handleMouseEnter: () -> Void = { () -> Void in
    if (self.hoverable) {
      self.isHovered = true;
    }
    var handleMouseLeave: () -> Void = { () -> Void in
      if (self.hoverable) {
        self.isHovered = false;
      }

      var body: some View {
        VStack(alignment: .leading, spacing: 8) {
          Show {
            Image("placeholder").resizable().aspectRatio(contentMode: .fit)
            // Dynamic styles not fully implemented
            .modifier(/* Dynamic style handling needed here */)"
          }
          Show {
            Text("")
            // Dynamic styles not fully implemented
            .modifier(/* Dynamic style handling needed here */)"
          }
          Show {
            Text("")
            // Dynamic styles not fully implemented
            .modifier(/* Dynamic style handling needed here */)"
          }
          Show {
            VStack(alignment: .leading, spacing: 8) {
              VStack(alignment: .leading, spacing: 8) {

              }
            }
          }
        }
        // Dynamic styles not fully implemented
        .modifier(/* Dynamic style handling needed here */)"
        .onMouseEnter(self.handleMouseEnter())
        .onMouseLeave(self.handleMouseLeave())
        .onTapGesture(self.onClick())

        }

        }

        #if DEBUG
        struct Card_Previews: PreviewProvider {
          static var previews: some View {
            Card(
              hoverable: /* provide preview value */,
              onClick: /* provide preview value */,
              className: /* provide preview value */,
              image: /* provide preview value */,
              title: /* provide preview value */,
              subtitle: /* provide preview value */,
              children: /* provide preview value */
            )
          }
        }
        #endif